### PR TITLE
Added Radius of Curvature Function to Firmware Polynomial

### PR DESCRIPTION
**Pull request by EvanMorcom**
### Description

Added an implementation of calculating the radius of curvature of a polynomial. 

This includes the macro and a few definitions.
### Testing Done

Edge-case unit test and a general test with one 3rd order polynomial

### Review Checklist

- [X] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [X] **Remove all commented out code**
- [X] **Remove extra print statements**: for example, those just used for testing
- [X] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [X] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

**Review by jonathanlew — DISMISSED** [review]
Assuming this is copied from trajectory planner and will replace that code when that PR is merge, then LGTM!

**Comment by EvanMorcom** [comment]
> Assuming this is copied from trajectory planner and will replace that code when that PR is merge, then LGTM!

Yep, I'm just breaking this one out 👍

**Review by MathewMacDougall — DISMISSED** [review]

**Review by garethellis0 — CHANGES_REQUESTED** [review]

**Review by MathewMacDougall — APPROVED** [review]

**Review by akhilveeraghanta — APPROVED** [review]
LGTM :fire:

File: src/firmware/shared/math/polynomial_2d.c [hunk 1]
```diff
@@ -195,6 +196,47 @@ GENERATE_2D_POLYNOMIAL_GET_T_VALUE_AT_ARC_LENGTH_FUNCTION_DEFINITION(3)
...
+            shared_polynomial1d_getValueOrder##ORDER_MINUS_ONE(first_deriv.x, s) *       \
+                shared_polynomial1d_getValueOrder##ORDER_MINUS_TWO(second_deriv.y, s) -  \
+            shared_polynomial1d_getValueOrder##ORDER_MINUS_ONE(first_deriv.y, s) *       \
+                shared_polynomial1d_getValueOrder##ORDER_MINUS_TWO(second_deriv.x, s));  \
+        const float denominator = pow(                                                   \
+            pow(shared_polynomial1d_getValueOrder##ORDER_MINUS_ONE(first_deriv.x, s),    \
+                2) +                                                                     \
+                pow(shared_polynomial1d_getValueOrder##ORDER_MINUS_ONE(first_deriv.y,    \
+                                                                       s),               \
+                    2),                                                                  \
```
[the remark this answers is not in the document]
Comment (garethellis0) [reply src/firmware/shared/math/polynomial_2d.c]: Yeah, sorry, your final interpretation was correct.

**Review by garethellis0 — APPROVED** [review]
LGTM :+1: :tada:
